### PR TITLE
Improve performance of Physics

### PR DESCRIPTION
**Pull request by WenhaiLin**
1.Move the logic of updating bodies' position and rotation to Node::updatePhysicsBodyTransform
2.Move the logic of coordinate conversion to Node::processParentFlags from PhysicsBody::update
3.Try to avoid the recursion

### CODE REVIEW

**Comment by Dhilan007** [comment]
[ci rebuild]

**Comment by WenhaiLin** [comment]
[ci rebuild]

File: cocos/2d/CCNode.cpp [hunk 1]
```diff
@@ -2128,52 +2004,65 @@ void Node::setPhysicsBody(PhysicsBody* body)
...
+    {
+        node->updatePhysicsBodyTransform(scene, _modelViewTransform, flags, scaleX, scaleY);
     }
 }
 
-PhysicsBody* Node::getPhysicsBody() const
+void Node::updateTransformFromPhysics(const Mat4& parentTransform, uint32_t parentFlags)
 {
-    return _physicsBody;
+    auto newPos = _physicsBody->getPosition();
```
Comment (minggo) [inline cocos/2d/CCNode.cpp]: Use auto&?

**Comment by minggo** [comment]
[ci rebuild]

File: cocos/2d/CCNode.cpp (hunk 1 above)
Comment (WenhaiLin) [inline cocos/2d/CCNode.cpp:2052]: It involved a compiler optimization technique. For more information: http://en.wikipedia.org/wiki/Return_value_optimization

**Pull request merged**